contracts/account/extensions/ERC7579Modules/SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,475 @@
+        }
+
+        for (uint256 i = 0; i < _guardians.length; i++) {
+            if (_guardians[i] == address(0)) {
+                revert InvalidGuardian();
+            }
+            bool added = _recoveryConfigs[account].guardians.add(_guardians[i]);
+            if (!added) {
+                revert InvalidGuardians();
+            }
+        }
+
+        _recoveryConfigs[account].threshold = _threshold;
+        _recoveryConfigs[account].timelock = _timelock;
+        _recoveryConfigs[account].recoveryStart = 0;
+        _recoveryConfigs[account].nonce = 0;
+
+        emit ModuleInstalledReceived(account, data);
+    }
+
+    /// @notice De-initializes the module, clearing all guardian configuration
+    /// @dev Called by the ERC-7579 Account during module uninstallation
+    /// @param data Additional data (unused)
+    function onUninstall(bytes calldata data) external override {
+        address account = msg.sender;
+
+        // Clear the guardians EnumerableSet
+        address[] memory guardians = _recoveryConfigs[account].guardians.values();
+        for (uint256 i = 0; i < guardians.length; i++) {
+            _recoveryConfigs[account].guardians.remove(guardians[i]);
+        }
+
+        // clear the recovery config
+        // slither-disable-next-line mapping-deletion
+        delete _recoveryConfigs[account];
+
+        emit ModuleUninstalledReceived(account, data);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                MODULE MAIN LOGIC
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Starts the recovery process for an account
+    /// @dev Requires threshold number of valid guardian signatures
+    /// @param account The account to start recovery for
+    /// @param guardianSignatures Array of guardian signatures authorizing the recovery
+    function startRecovery(
+        address account,
+        GuardianSignature[] calldata guardianSignatures
+    ) external whenRecoveryIsNotStarted(account) {
+        if (guardianSignatures.length < _recoveryConfigs[account].threshold) {
+            revert ThresholdNotMet();
+        }
+        if (!guardianSignaturesAreValid(account, guardianSignatures)) {
+            revert InvalidGuardianSignatures();
+        }
+
+        // increment nonce.
+        _recoveryConfigs[account].nonce++;
+
+        // set recovery start time.
+        _recoveryConfigs[account].recoveryStart = block.timestamp;
+
+        emit RecoveryStarted(account);
+    }
+
+    /// @notice Executes the recovery process after timelock period
+    /// @dev Can only execute calls to reconfigure installed Validator Modules
+    /// @param account The account to execute recovery for
+    /// @param guardianSignatures Array of GuardianSignature authorizing the recovery
+    /// @param executionCalldata The encoded call to reconfigure a Validator Module
+    function executeRecovery(
```

Comment:
I looks like there is a massive security flaw here, the signature does NOT bind to the executionCalldata !!!

If I intercept the tx in the memory pool, can I front run it with a different `executionCalldata`, and use the signature from the guardians to recover to my benefit ? I don't see what is stopping this from happening.

Overall I think the design should have the guardian sign what the recovery will do, and commit that when doing `startRecovery`. That way, its clear what is going to be done.

The execute should probably not need signatures, since they were already provided at the begining. And the guardians should be able to change their mind and cancel if they realise their recovery is not good.